File: Napakalaki Java/NapakalakiGame/Monster.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package NapakalakiGame;


/**
 *
 * @author antoniomj
 */
public class Monster{
    
    private String name;
    private int combatLevel;
    private Prize premioMonstruo;
    private BadConsequence malRollo;
    private int levelChangeAgainstCultistPlayer;
    
    public Monster(String Name, int CombatLevel, BadConsequence MalRollo, Prize PremioMonstruo){
        
        name = Name;
        combatLevel = CombatLevel;
        malRollo = MalRollo;
        premioMonstruo = PremioMonstruo;
        levelChangeAgainstCultistPlayer = 0;
        
    }
    
    public Monster(String Name, int CombatLevel, BadConsequence MalRollo, Prize PremioMonstruo, int levelChange){
        
        name = Name;
        combatLevel = CombatLevel;
        malRollo = MalRollo;
        premioMonstruo = PremioMonstruo;
        levelChangeAgainstCultistPlayer = levelChange;
        
    }
    
    public int getCombatLevelAgainstCultistPlayer() {
        return levelChangeAgainstCultistPlayer + combatLevel;
    }
    
    public String Getname(){
        
        return name;
    }
    
    public int GetCombatLevel(){
        
        return combatLevel;
    }
    
    public BadConsequence GetMalRollo(){
        
        return malRollo;
        
    }
    
    public Prize GetPremio(){
        
        return premioMonstruo;
    }
    public String toString(){
        
        return "Nombre del monstruo: " + name + "\nNivel del monstruo: " 
                + Integer.toString(combatLevel)
                + "\nPremio del monstruo: " + premioMonstruo.toString() +
                "\nMal Rollo: " + malRollo.toString(); 
        
    }
    
    public void SetMalRollo(BadConsequence mal_rollo){
        
        malRollo = mal_rollo;
        
    }
    
    public int getLevelsGained(){
        int niveles;
        niveles = premioMonstruo.Getlevel();
     
        return niveles;
    }
    
    public int getTreasuresGained(){
        int tesoros;
        tesoros = premioMonstruo.GetTreasures();
     
        return tesoros;
    }
}

File: Napakalaki Ruby/Monster.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative 'BadConsequence.rb'
require_relative 'NumericBadConsequence.rb'
require_relative 'DeathBadConsequence.rb'
require_relative 'SpecificBadConsequence.rb'
require_relative 'Prize.rb'

module NapakalakiGame

class Monster
 
  attr_reader :name, :combat_level, :bad_consequence, :prize, :levelChangeAgainstCultistPlayer;

  def initialize(nombre, nivel, mal_rollo, premio,nivel_aux_sectarios)
    
    @name = nombre
    @combat_level = nivel
    @bad_consequence = mal_rollo
    @prize = premio
    @levelChangeAgainstCultistPlayer = nivel_aux_sectarios
    
  end
  
  def to_s ()
    
    return "Nombre del bicho: #{@name}\nNivele del bicho #{@combat_level}\n" + @bad_consequence.to_s +
            "\n" + @prize.to_s + "\n"
    
  end
  
  def getLevelGained
    return @prize.getLevels
  end
  
  def getTreasuresGained
    return @prize.getTreasures
  end
  
  def getCombatLevel
    return  @combat_level
  end
  
  def getCombatLevelAgainstCultistPlayer
    aux = @combat_level + @levelChangeAgainstCultistPlayer
    return aux
  end
  
  def getBadConsequence
    return @bad_consequence
  end
  
  def getName
    return @name
  end
  
end

end
